### PR TITLE
add php8.1 support

### LINKED ISSUE #12: PHP 8.1 compatibility
Hey there. Thanks for the cool tool.
It seems the Ploi CLI is not compatible with PHP8.1. I get the following error:

```
Deprecated: Return type of HumbugBox3130\KevinGH\RequirementChecker\RequirementCollection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/.box/src/RequirementCollection.php on line 12

Deprecated: Return type of HumbugBox3130\KevinGH\RequirementChecker\RequirementCollection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/.box/src/RequirementCollection.php on line 16

Deprecated: Return type of Illuminate\Container\Container::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/illuminate/container/Container.php on line 1405

Deprecated: Return type of Illuminate\Container\Container::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/illuminate/container/Container.php on line 1416

Deprecated: Return type of Illuminate\Container\Container::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/illuminate/container/Container.php on line 1428

Deprecated: Return type of Illuminate\Container\Container::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/illuminate/container/Container.php on line 1441

Deprecated: Return type of PhpOption\Some::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/phpoption/phpoption/src/PhpOption/Some.php on line 152

Deprecated: Return type of PhpOption\None::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/phpoption/phpoption/src/PhpOption/None.php on line 118

Deprecated: Return type of Illuminate\Config\Repository::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/illuminate/config/Repository.php on line 141

Deprecated: Return type of Illuminate\Config\Repository::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/illuminate/config/Repository.php on line 152

Deprecated: Return type of Illuminate\Config\Repository::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/illuminate/config/Repository.php on line 164

Deprecated: Return type of Illuminate\Config\Repository::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/illuminate/config/Repository.php on line 175

Deprecated: Return type of Symfony\Component\Finder\Finder::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/symfony/finder/Finder.php on line 608

Deprecated: Return type of Symfony\Component\Finder\Finder::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/symfony/finder/Finder.php on line 690

Deprecated: Return type of Symfony\Component\Finder\Iterator\FileTypeFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/symfony/finder/Iterator/FileTypeFilterIterator.php on line 42

Deprecated: Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 55

Deprecated: Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::hasChildren() should either be compatible with RecursiveIterator::hasChildren(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 74

Deprecated: Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::getChildren() should either be compatible with RecursiveIterator::getChildren(): ?RecursiveIterator, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 82

Deprecated: Return type of Symfony\Component\Finder\Iterator\FilenameFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/symfony/finder/Iterator/FilenameFilterIterator.php on line 28

Deprecated: Return type of Symfony\Component\Finder\Iterator\PathFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/robdekort/.composer/vendor/schmidfelix/ploi-cli/builds/ploi/vendor/symfony/finder/Iterator/PathFilterIterator.php on line 27
```

### PR DESCRIPTION
**Pull request by fheider**
fixes #12

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,16 +16,16 @@
         }
     ],
     "require": {
-        "php": "^7.4|^8.0"
+        "php": "^8.0",
+        "laravel-zero/framework": "^9.0",
+        "guzzlehttp/guzzle": "^7.4",
+        "illuminate/http": "^9.0",
```
Comment (schmidfelix) [inline composer.json:23]: Sorry for the late answer, I've completely missed it. Thanks for the PR, but I'm just wondering if we need all those dependencies in a production build or if they should be enough as dev dependency?

**Comment by EQuimper** [comment]
Any idea when this feature can be add :) I have the same issue with php 8.1 :(

**Comment by chasegiunta** [comment]
@fheider @schmidfelix hoping we can get this merged soon 🙏

**Comment by fheider** [comment]
@schmidfelix i updated the dependencies

**Pull request merged**